provider_client.go

```diff
@@ -179,6 +189,19 @@ func (client *ProviderClient) Request(method, url string, options RequestOpts) (
 		return nil, err
 	}
 
+	if resp.StatusCode == 401 {
```

Comment:
You could also use the constant here to avoid the magic number; `http.StatusUnauthorized`.